homeassistant/components/tplink/__init__.py

```diff
@@ -116,8 +127,11 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     hass.data.setdefault(DOMAIN, {})
 
     async def _async_discovery(*_: Any) -> None:
-        if discovered := await async_discover_devices(hass):
-            async_trigger_discovery(hass, discovered)
+        credentials = await get_credentials(hass)
```

Comment:
unrelated to this PR but there are some issues with `get_credentials`

- nothing is awaited in `get_credentials` or `set_credentials`. They should be an async callback functions instead
- its unexpected that its a dict and not a `Credentials` object so a new `Credentials` gets created every time. This may be intentional though